addons/music/fnc_compileMusic.sqf

```diff
@@ -0,0 +1,40 @@
+/* ----------------------------------------------------------------------------
+Function: CBA_fnc_compileMusic
+Description:
+A function used to gather a list of all usable music.
+Parameters:
+none
+Example:
+(begin example)
+_allmusic = [] call CBA_fnc_compileMusic
+(end example)
+Returns:
+config
+Author:
+Fishy
+---------------------------------------------------------------------------- */
+#include "script_component.hpp"
+if (isNil QGVARMAIN(compiledMusic)) then {
+_config = configFile >> 'CfgMusic';
+_missionConfig = missionConfigFile >> 'CfgMusic';
+_unsortedSongs = [];
+for [{_i=0}, {_i < (count _config)}, {INC(_i)}] do {
+_song = _config select _i;
+_songDuration = getNumber (_song >> 'duration');
+if (_songDuration !=0) then {
```

Comment:
Could be better to use `(_songDuration > 0)` to prevent a value of e.g `-1`
otherwise:
`!(_songDuration isEqualTo 0)`